File: 2020/d1/python/day1.py
def get_challenge1_answer(entries):
    for i in range(0, len(entries)):
        for j in range(i + 1, len(entries)):
            if entries[i] + entries[j] == 2020:
                return entries[i] * entries[j]


def get_challenge2_answer(entries):
    for i in range(0, len(entries)):
        for j in range(i + 1, len(entries)):
            for k in range(j + 1, len(entries)):
                if entries[i] + entries[j] + entries[k] == 2020:
                    return entries[i] * entries[j] * entries[k]


f = open('2020\\d1\\input.txt')
entries = list(map(lambda x: int(x.strip()), f.readlines()))
answer1 = get_challenge1_answer(entries)
answer2 = get_challenge2_answer(entries)

print(answer1)
print(answer2)

File: 2020/d1/julia/day1.jl
function find_pair_sum_to(data, n)
    for x in data
        rem = n - x
        if rem in data
            return (x, rem)
        end
    end
	return (nothing, nothing)
end

function find_triplet(data, n)
    for x in data
        rem = n - x
        (a, b) = find_pair_sum_to(data, rem)
		if a != b && a != x && b != x
        	return (a, b, x)
		end
    end
end

data = open("2020\\d1\\input.txt") do f
    parse.(Int32, readlines(f))
end

a1 = find_pair_sum_to(data, 2020)
a2 = find_triplet(data, 2020)

println("$a1 -> $(prod(a1))")
println("$a2 -> $(prod(a2))")